connectors/src/connectors/github/index.ts

```diff
@@ -63,6 +64,41 @@ export async function createGithubConnector(
   }
 }
 
+export async function updateGithubConnector(
+  connectorId: ModelId,
+  connectionId: string
+): Promise<Result<string, ConnectorsAPIErrorResponse>> {
+  const c = await Connector.findOne({
+    where: {
+      id: connectorId,
+    },
+  });
+  if (!c) {
+    logger.error({ connectorId }, "Connector not found");
+    return new Err({
+      error: {
+        message: "Connector not found",
+        type: "connector_not_found",
+      },
+    } as ConnectorsAPIErrorResponse);
+  }
+
+  const oldGithubInstallationId = c.connectionId;
+  const newGithubInstallationId = connectionId;
+
+  if (oldGithubInstallationId !== newGithubInstallationId) {
+    return new Err({
+      error: {
+        type: "connector_update_unauthorized",
```

Comment:
Actually... I am updating the `updateXxxConnector` functions to accept other params. This error naming is quite generic, and you are catching it from the front -- should we pick a more specific error type to avoid confusion ?

Reply:
Agreed 👍🏻